src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitQueue.java

```diff
@@ -1,61 +1,69 @@
  *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
  */
 
 package com.exactpro.th2.common.schema.message.impl.rabbitmq;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Objects;
-import java.util.concurrent.atomic.AtomicReference;
-
-import org.jetbrains.annotations.NotNull;
-
+import com.exactpro.th2.common.schema.message.FilterFunction;
 import com.exactpro.th2.common.schema.message.MessageQueue;
 import com.exactpro.th2.common.schema.message.MessageSender;
 import com.exactpro.th2.common.schema.message.MessageSubscriber;
 import com.exactpro.th2.common.schema.message.configuration.QueueConfiguration;
 import com.exactpro.th2.common.schema.message.impl.rabbitmq.connection.ConnectionManager;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
 
 public abstract class AbstractRabbitQueue<T> implements MessageQueue<T> {
 
     private final AtomicReference<ConnectionManager> connectionManager = new AtomicReference<>();
     private final AtomicReference<QueueConfiguration> queueConfiguration = new AtomicReference<>();
+    private final AtomicReference<FilterFunction> filterFunc = new AtomicReference<>();
 
     private final AtomicReference<MessageSender<T>> sender = new AtomicReference<>();
     private final AtomicReference<MessageSubscriber<T>> subscriber = new AtomicReference<>();
 
     @Override
     public void init(@NotNull ConnectionManager connectionManager, @NotNull QueueConfiguration queueConfiguration) {
+        this.init(connectionManager, queueConfiguration, FilterFunction.DEFAULT_FILTER_FUNCTION);
+    }
+
+    @Override
+    public void init(@NotNull ConnectionManager connectionManager, @NotNull QueueConfiguration queueConfiguration, @NotNull FilterFunction filterFunc) {
         Objects.requireNonNull(connectionManager, "Connection can not be null");
         Objects.requireNonNull(queueConfiguration, "Queue configuration can not be null");
+        Objects.requireNonNull(filterFunc, "Filter function can not be null");
 
-        if (this.connectionManager.get() != null && this.queueConfiguration.get() != null) {
+        if (this.connectionManager.get() != null && this.queueConfiguration.get() != null && this.filterFunc.get() != null) {
```

Comment:
Do you mean `A || B || C`?

Reply:
Extract this logic to separate function and call at begin of public methods.

Reply:
I think extracted the logic to a separate function is a bad idea. Because we can not call it in `getSubscriber` and `getSender`, because in ones we get values from fields, but in check function, we again check fields values, which can change.

Reply:
It is not actually because we migrated from init methods to constructor in another branch.